caddyhttp/httpserver/replacer.go

```diff
@@ -132,6 +147,36 @@ func NewReplacer(r *http.Request, rr *ResponseRecorder, emptyValue string) Repla
 	return rep
 }
 
+func canLogRequest(r *http.Request) (canLog bool) {
+	if r.Method == "POST" || r.Method == "PUT" {
+		for _, cType := range r.Header[headerContentType] {
+			// the cType could have charset and other info
+			if strings.Index(cType, contentTypeJSON) > -1 || strings.Index(cType, contentTypeXML) > -1 {
+				canLog = true
+				break
+			}
+		}
+	}
+	return
+}
+
+// readRequestBody reads the request body and sets a
+// new io.ReadCloser that has not yet been read.
+func readRequestBody(r *http.Request, n int64) ([]byte, error) {
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, n))
```

Comment:
I'm not sure it is the right fit for this scenario.  Here we just want to log a portion of the body (100 KB).  If the request's body is less then 100 KB then all is well and we'll get the full body size.

Looking at the implementation of the MaxBytesReader, it wants a ResponseWriter to set the status code `Request Too Large` if request's body size is less than the 100 KB limit that we're trying to read.